docs/design/2022-04-15-multi-schema-change.md

```diff
@@ -0,0 +1,153 @@
+# Proposal: Support Multi-Schema Change
+
+- Author(s): [tangenta](https://github.com/tangenta)
+- Tracking Issue: https://github.com/pingcap/tidb/issues/14766
+
+## Abstract
+
+This proposes an implementation of Multi-Schema Change.
+
+## Background
+
+Multi-Schema Change refers to defining multiple schema changes in one SQL statement, including column and index `ADD` , `ALTER` , `DROP` and `CHANGE` , as well as table option changes. For example:
+
+```sql
+CREATE TABLE t (a INT, c INT);
+ALTER TABLE t ADD COLUMN b INT AUTO_INCREMENT PRIMARY KEY, 
+ MODIFY COLUMN c CHAR(5),
+ ADD INDEX idx(a),
+ AUTO_INCREMENT = 1000;
+```
+
+Currently, TiDB only supports one schema change per SQL statement, and multi-schema changes in few cases. 
+
+When users try to migrate data from MySQL-like databases, they have to spend extra effort to rewrite a multi-schema change DDL to several single-schema change DDL. For the users who rely on ORM frameworks like flyway to construct SQLs automatically, rewriting SQL is not even possible.
+
+Above all, Multi-Schema Change can be a blocking issue for POC procedure.
+
+## Features
+
+Multi-Schema Change is one of MySQL's extended features to the SQL standard. It has the following features:
+
+- Atomicity: Schema changes from the same SQL statement either all succeed or all fail. In the above example, adding column b, modifying column c, adding the index of column a, and modifying the auto-increment ID of the table must all succeed or fail. The intermediate states are invisible to users.
+
+- Cascading: Whether the previously generated object in the same statement can be referenced for the schema object currently being changed.
+    ```sql
+    CREATE TABLE t (a INT);
+    ALTER TABLE t ADD COLUMN b INT, ADD INDEX idx(b);
+    -- Query OK, 0 rows affected (0.07 sec)
+    ALTER TABLE t ADD COLUMN c INT, ADD COLUMN d INT as (c+1);
```

Comment:
Do we decided to support those SQLs now?

Reply:
No. I have mentioned it in the "Compatibility" part.